lib/web_ui/lib/src/engine/util.dart

```diff
@@ -45,15 +45,27 @@ typedef Callbacker<T> = String? Function(Callback<T> callback);
 ///   return _futurize(_doSomethingAndCallback);
 /// }
 /// ```
+// Note: this function is not directly tested so that it remains private, instead an exact
+// copy of it has been inlined into the test at lib/ui/fixtures/ui_test.dart. if you change
+// this function, then you  must update the test.
 Future<T> futurize<T>(Callbacker<T> callbacker) {
   final Completer<T> completer = Completer<T>.sync();
-  final String? error = callbacker((T t) {
+  // If the callback synchronously throws an error, then synchronously
+  // rethrow that error instead of adding it to the completer. This
+  // prevents the Zone from receiving an uncaught exception.
+  bool sync = true;
+  final String? error = callbacker((T? t) {
     if (t == null) {
-      completer.completeError(Exception('operation failed'));
+      if (sync) {
```

Comment:
Took me a while to parse this code, wowza!

Reply:
It's a copy of the mobile version, so I'm reluctant to touch it. But I did add tests! :)

Reply:
The side effect on `sync` for async functions made me ponder this for a while :P